src/sentry/api/serializers/models/plugin.py

```diff
@@ -77,18 +90,9 @@ def serialize(self, obj, attrs, user):
         if obj.author:
             d["author"] = {"name": str(obj.author), "url": str(obj.author_url)}
 
-        # TODO(Leander): Convert this field to:
-        # d["isDeprecated"] = obj.slug in SHADOW_DEPRECATED_PLUGINS or datetime.today() > deprecation_date
-        # but we are late on deprecating right now
-        d["isDeprecated"] = obj.slug in SHADOW_DEPRECATED_PLUGINS
-
-        d["isHidden"] = (
-            not features.has(
-                SHADOW_DEPRECATED_PLUGINS.get(obj.slug), getattr(self.project, "organization", None)
-            )
-            if d["isDeprecated"]
-            else d.get("enabled", False) is False and obj.is_hidden()
-        )
+        d["isDeprecated"] = is_plugin_deprecated(obj, self.project)
```

Comment:
Since we rely on `is_plugin_deprecated` it is no serialized based on the org requesting it, and is not a 'source of truth'. Orgs flagged into the shadow deprecation will receive `False`, other orgs  will receive `True`. This prevents having to keep the `SHADOW_DEPRECATED_PLUGINS` const on both the frontend and backend (hence the deletion in `navigationConfiguration.tsx`).